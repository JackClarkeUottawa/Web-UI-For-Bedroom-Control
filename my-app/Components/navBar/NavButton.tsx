import React from "react";
import Link from "next/link";
interface navButtonProps {
  href?: string;
  text?: string;
}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface navButtonState {}
export default class NavButton extends React.Component<
  navButtonProps,
  navButtonState
> {
  render(): JSX.Element {
    if (!this.props.href) {
      //throw new Error("Unexpected error: missing link");
    }
    return (
      <Link href={this.props.href}>
        <a className="nav-link">{this.props.text}</a>
      </Link>
    );
  }
}
