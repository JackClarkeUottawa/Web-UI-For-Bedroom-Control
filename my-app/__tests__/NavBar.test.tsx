//Author: Jack Clarke
import React from "react";
import NavButton from "../Components/navBar/NavButton";
import renderer from "react-test-renderer";



it("button renders correctly", () => {
  const tree = renderer
    .create(<NavButton href="/" text="I like to eat" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
