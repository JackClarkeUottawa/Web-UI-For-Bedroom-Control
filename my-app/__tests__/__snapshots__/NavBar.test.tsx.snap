// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="nav-link"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  I like to eat
</a>
`;
